### DISCUSSION TITLE
“file” subproblem multiple files?

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#943] [Q&A] [answered]
Seemingly not the best practice; however, one of the key reasons we're trying to adopt this was that students could upload files (source code) for question types on some examples. Is it possible for the students to upload multiple files as part of a single submission/task? We have an existing system that the hope is this will replace where students can upload multiple files like both .cpp/.h or .c/.h files for a project that are the bits of a project students build. 

The students, other faculty, and probably myself will not want to switch if I can replicate this functionality for them. The documentation indicates, "Students should only have to upload files they modified." The only example I can find is in the [competitive programming example](https://github.com/UCL-INGI/competitive-programming/tree/master):

```yaml
problems:
    file:
        type: file
        allowed_exts:
        - .java
        - .cpp
        - .py
```
But this is based on the experience on https://inginious.org/ seems to only be a singular file, I didn't find any multi-file examples. Hoping it's just undocumented, but I thought I'd ask to see if anyone knows or knows where the documentation is. Hoping I don't have to extend or write my own plugin as not sure I'll have time for that before this summer or over the summer to get it working.

**Comment by anthonygego** [comment, marked as answer]
You can add as many subproblems as you want in a single task (that is, including file subproblems). Then you can virtually do whatever you want to with these several inputs in the grading scripts once you have fetched their content with ``getinput <problemid>``.

**Reply by javawolfpack** [reply]
That wasn't clear; thanks for the input will give it a go once I have an instance up to test. May follow this up with an example task if I get one working for others who have this question in the future.
